src/item_location.cpp

```diff
@@ -31,7 +33,12 @@ class item_location::impl
         virtual void remove_item() = 0;
 
     protected:
-        item *what = nullptr;
+        impl( item_uid &&id ) : id( std::move( id ) ) {}
```

Comment:
What is this move for?
In general every use of std::move() should be documented with a rationale.